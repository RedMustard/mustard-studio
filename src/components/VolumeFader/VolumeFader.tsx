import { h, Fragment } from 'preact';
import { MAX_VOLUME, MIN_VOLUME, VOLUME_STEPS } from '../../constants';

interface VolumeFaderProps {
    value: number,
    classSuffix?: string;
    onChange: (value: number) => void;
}

export const VolumeFader = ({
    value,
    onChange,
    classSuffix = '',
}: VolumeFaderProps) => (
    <Fragment>
        <input
            className={`volume-fader ${classSuffix ? `${classSuffix}-volume-fader` : ''}`}
            onInput={(event) => onChange(parseFloat(event.currentTarget.value))}
            list="volumes"
            max={MAX_VOLUME}
            min={MIN_VOLUME}
            name={`${classSuffix ? `${classSuffix}-volume` : 'volume'}`}
            step={VOLUME_STEPS}
            type="range"
            value={value}
        />
        <datalist id="volumes">
            <option value={MIN_VOLUME} label="Mute" />
            <option value={MAX_VOLUME} label="100%" />
        </datalist>
    </Fragment>
);
