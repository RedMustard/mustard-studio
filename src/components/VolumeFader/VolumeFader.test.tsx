import { h } from 'preact';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { VolumeFader } from './VolumeFader';


const baseProps = {
    value: 1.0,
    onInput: jest.fn(),
    onMouseDown: jest.fn(),
    classSuffix: 'foo',
};

describe('<VolumeFader />', () => {
    it('renders with basic props', () => {
        const wrapper = shallow(
            <VolumeFader
                {...baseProps}
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('calls onInput when changed', () => {
        const wrapper = mount(
            <VolumeFader
                {...baseProps}
            />,
        );
        wrapper.find('input').simulate('input');
        expect(baseProps.onInput).toBeCalled();
    });

    it('calls onMouseDown when clicked', () => {
        const wrapper = mount(
            <VolumeFader
                {...baseProps}
            />,
        );
        wrapper.find('input').simulate('mousedown');
        expect(baseProps.onInput).toBeCalled();
    });
});
