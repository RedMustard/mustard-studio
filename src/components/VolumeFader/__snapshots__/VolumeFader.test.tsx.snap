// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VolumeFader /> renders with basic props 1`] = `
Array [
  <input
    className="volume-fader "
    list="volumes"
    max={1}
    min={0}
    name="volume"
    onInput={[Function]}
    step={0.01}
    type="range"
    value={1}
  />,
  <datalist
    id="volumes"
  >
    <option
      label="Mute"
      value={0}
    />
    <option
      label="100%"
      value={1}
    />
  </datalist>,
]
`;

exports[`<VolumeFader /> renders with custom class suffix 1`] = `
Array [
  <input
    className="volume-fader foo-volume-fader"
    list="volumes"
    max={1}
    min={0}
    name="foo-volume"
    onInput={[Function]}
    step={0.01}
    type="range"
    value={1}
  />,
  <datalist
    id="volumes"
  >
    <option
      label="Mute"
      value={0}
    />
    <option
      label="100%"
      value={1}
    />
  </datalist>,
]
`;
